client/v2/internal/util/util.go

```diff
@@ -24,3 +36,52 @@ func ResolveMessageType(resolver protoregistry.MessageTypeResolver, descriptor p
 
 	return dynamicpb.NewMessageType(descriptor)
 }
+
+// IsSupportedVersion is used to determine in which version of a module / sdk a rpc was introduced.
+// It returns false if the rpc has comment for an higher version than the current one.
+func IsSupportedVersion(input string) bool {
+	return isSupportedVersion(input, buildInfo)
+}
+
+// isSupportedVersion is used to determine in which version of a module / sdk a rpc was introduced.
+// It returns false if the rpc has comment for an higher version than the current one.
+// It takes a buildInfo as argument to be able to test it.
+func isSupportedVersion(input string, buildInfo *debug.BuildInfo) bool {
+	if input == "" || buildInfo == nil {
+		return true
+	}
+
+	moduleName, version := parseSinceComment(input)
+	for _, dep := range buildInfo.Deps {
+		if !strings.Contains(dep.Path, moduleName) {
+			continue
+		}
+
+		return version <= dep.Version
+	}
+
+	return true // if cannot find the module consider it's supported
+}
+
+// parseSinceComment parses the `// Since: cosmos-sdk v0.xx` comment on rpc.
+func parseSinceComment(input string) (string, string) {
+	var (
+		moduleName string
+		version    string
+	)
+
+	input = strings.ToLower(input)
+	input = strings.ReplaceAll(input, "cosmos sdk", "cosmos-sdk")
+
+	re := regexp.MustCompile(`\/\/\s*since: (\S+) (\S+)`)
+	matches := re.FindStringSubmatch(input)
+	if len(matches) >= 3 {
+		moduleName, version = matches[1], matches[2]
+
+		if !strings.Contains(version, "v") {
```

Comment:
"0.17.5-veronica" would fail. I believe you instead want to use strings.HasPrefix instead